packages/components/src/date-time/time.js

```diff
@@ -23,347 +28,258 @@ import TimeZone from './timezone';
  */
 const TIMEZONELESS_FORMAT = 'YYYY-MM-DDTHH:mm:ss';
 
-class TimePicker extends Component {
-	constructor() {
-		super( ...arguments );
-		this.state = {
-			day: '',
-			month: '',
-			year: '',
-			hours: '',
-			minutes: '',
-			am: '',
-			date: null,
-		};
-		this.changeDate = this.changeDate.bind( this );
-		this.updateMonth = this.updateMonth.bind( this );
-		this.onChangeMonth = this.onChangeMonth.bind( this );
-		this.updateDay = this.updateDay.bind( this );
-		this.onChangeDay = this.onChangeDay.bind( this );
-		this.updateYear = this.updateYear.bind( this );
-		this.onChangeYear = this.onChangeYear.bind( this );
-		this.updateHours = this.updateHours.bind( this );
-		this.updateMinutes = this.updateMinutes.bind( this );
-		this.onChangeHours = this.onChangeHours.bind( this );
-		this.onChangeMinutes = this.onChangeMinutes.bind( this );
-		this.renderMonth = this.renderMonth.bind( this );
-		this.renderDay = this.renderDay.bind( this );
-		this.renderDayMonthFormat = this.renderDayMonthFormat.bind( this );
-	}
+/**
+ * <NumberInput>
+ * A shared component to parse, validate, and handle remounting of the underlying <input> element
+ *
+ * @param {Object} props Component props
+ * @param {string} props.as Render the component as specific element tag, defaults to "input"
+ * @param {number | string} props.value The default value of the component
```

Comment:
Maybe this should allow any type given the component can be rendered as anything that accepts a value prop.